flake.nix

```diff
@@ -0,0 +1,49 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    crane.url = "github:ipetkov/crane";
+    crane.inputs.nixpkgs.follows = "nixpkgs";
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, crane, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
```

Comment:
I think it'd be cleaner to avoid `flake-utils` and be explicit about what systems are supported. Preferably replacing this with [flake-parts](https://github.com/hercules-ci/flake-parts)

https://ayats.org/blog/no-flake-utils/